hazelcast/src/main/java/com/hazelcast/sql/impl/exec/sort/SortKeyComparator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.exec.sort;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Comparator fot the sort key.
+ */
+@SuppressFBWarnings(value = "SE_COMPARATOR_SHOULD_BE_SERIALIZABLE", justification = "Never serialized")
+public class SortKeyComparator implements Comparator<SortKey> {
+    /** List of ascending collations. */
+    private final List<Boolean> ascs;
```

Comment:
`boolean[]` could be a better choice from the performance perspective since this is a pretty sensitive place.

Reply:
Improved in https://github.com/hazelcast/hazelcast/pull/18013/commits/6c035fbc48defd9eacd3e99875041760c448b7c0